packages/codegen/src/generateEntityCodegenFile.ts

```diff
@@ -594,7 +594,11 @@ function fieldHasDefaultValue(field: PrimitiveField | EnumField): boolean {
   );
 }
 
-function generatePolymorphicTypes(meta: EntityDbMetadata) {
+function generatePolymorphicTypes(meta: EntityDbMetadata, baseEntity?: EntityDbMetadata) {
+  // If this entity has a base class we will filter out any polymorphic fields that are from the base class
+  if (meta.baseClassName && baseEntity) {
+    meta.polymorphics = meta.polymorphics.filter((p) => !baseEntity.polymorphics.includes(p));
+  }
```

Comment:
@0ces just from reading the comment, it sounds like we were like including these fields in the output twice?

Like if `BasePublisher` had a poly field, the `SubPublisher` would also output it, and then that would break somehow? 

Would love to see some test coverage--looks like you were working on that, but then reverted?

Reply:
Yeah, I set that in case in the `joist-config.json` we set something like:
```json
{
...
	"SubPub

Reply:
Yeah, I set that in case in the `joist-config.json` we set something like this:

```json
{
	...
	"BasePublisher": { "relations": { "parent": { "polymorphic": true } }, "tag": "bp" },
	"SubPublisher": { "relations": { "parent": { "polymorphic": true } }, "tag": "sp" },
	...
}
```

It would just take the base poly, but it might be worth to just keep the child polymorphic, now that I think about it, it might be useful to be able to overwrite the polys from the base (?), I'll remove this and update the logic I added at the PolymorphicReference to select the correct one

Yes, I reverted the tests as there were more changes than I expected, I'll find another example instead of "ReviewComment", I was testing with the some BP-GQL-Server entities and it worked as expected but I'll test this here better

I'll convert this PR into a draft for now and will ask for a review when this gets into a better shape

Reply:
Hm, "overwrite polys from the base", do you mean something like:

* `BasePublisher.parent = Foo | Bar` (because of `base_publishers.parent_foo_id` and `base_publishers.parent_bar_id`), but
* `SubPublisher.parent = Foo | Bar | Zaz` (because of a new `sub_publishers.parent_zaz_id` that is a FK only added in the sub publisher table?

I could see that working, albeit it sounds kinda complicated. :-) But doable. What's the use case we're trying to use this over in graphql-service?

Reply:
Yes, I don't see an example of overwriting the polymorphic relationship, but in the graphql-service we created a sub-entity for ToDos and the sub-entity's parent polymorphic feature wasn't working